modules/spirv_cross/SCsub

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+
+Import("env")
+Import("env_modules")
+
+env_spirv = env_modules.Clone()
+
+# Thirdparty source files
+
+thirdparty_obj = []
+
+if env["metal"]:
+    thirdparty_dir = "#thirdparty/spirv-cross/"
+    thirdparty_sources = [
+        "spirv_cfg.cpp",
+        "spirv_cross_util.cpp",
+        "spirv_cross.cpp",
+        "spirv_parser.cpp",
+        "spirv_msl.cpp",
+        "spirv_reflect.cpp",
+        "spirv_glsl.cpp",
+        "spirv_cross_parsed_ir.cpp",
+        "spirv_cpp.cpp",
+    ]
+
+    thirdparty_sources = [thirdparty_dir + file for file in thirdparty_sources]
+
+    env_spirv.Prepend(CPPPATH=["#thirdparty/spirv-cross/"])
+
+    env_thirdparty = env_spirv.Clone()
+    env_thirdparty.disable_warnings()
+    # Must enable exceptions for SPIRV-Cross; otherwise, it will abort the process on errors
+    if "-fno-exceptions" in env_thirdparty["CXXFLAGS"]:
+        env_thirdparty["CXXFLAGS"].remove("-fno-exceptions")
+    env_thirdparty.Append(CXXFLAGS=["-fexceptions"])
```

Comment:
That's for a future PR, but it would be nice to look into whether we can patch SPIRV-Cross to avoid relying on exceptions.

As I mentioned, Godot doesn't use exceptions anywhere, and disabling them reduces the binary size (and IIRC compilation time) signficantly.

Reply:
> As I mentioned, Godot doesn't use exceptions anywhere, and disabling them reduces the binary size (and IIRC compilation time) significantly.

Hopefully scoping the exceptions to spriv-cross and the Metal driver, it shouldn't impact build times or code-size significantly. I could even scope the Metal drive exception use to a single code unit?